### DISCUSSION TITLE
Does GLSP only support svg element view?

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#925] [Q&A]
Hi,
    I want to create a custom view for my node, for example my custom is a simple div. but  I get error
![image](https://user-images.githubusercontent.com/4537809/219827970-5dcc7c45-5ff1-4371-9ba8-1b2112aff3c5.png)
![image](https://user-images.githubusercontent.com/4537809/219827988-ef1f949a-59f0-430d-a9a6-29de2d55914d.png)
Dose IView only support <g></g> like svg render element? If I want to add some HTML element or react component to my custom view. is it possible?

**Comment by planger** [comment]
Hi,

I don't think it is possible to directly use HTML as your markup will be embedded directly in SVG `g` elements inside an `svg` element. For a use case like yours, we typically use either [UI extensions](https://www.eclipse.org/glsp/documentation/ui-extensions/) to render HTML on top of the diagram or [SVG's foreign objects](https://www.eclipse.org/glsp/documentation/rendering/#foreignobjectviewhttpsgithubcomeclipsesprottyblobmasterpackagessprottysrclibgeneric-viewstsx).

For more information on how to use those approaches, follow the links above.

Best wishes,

Philip

**Reply by wss29** [reply]
Hi,
   Thank you for your reply, SVG foreign object method seems fit to me, but the above link is not available. could you send an available one?
![image](https://user-images.githubusercontent.com/4537809/219870594-fc59d24d-abe9-4f71-802f-e4e4e2236801.png)

**Reply by wss29** [reply]
I try to implement a foreign object, but the div not render
![image](https://user-images.githubusercontent.com/4537809/219872438-f84503d4-4be7-43a7-be37-1b1172a0b25d.png)
![image](https://user-images.githubusercontent.com/4537809/219872466-cd94636e-ca42-49af-be70-cae5c71fae87.png)
![image](https://user-images.githubusercontent.com/4537809/219872483-deee11bf-f2d0-4e84-82f6-3cbbc5329509.png)
![image](https://user-images.githubusercontent.com/4537809/219872508-3939036f-36f0-442c-ae2d-02116af40582.png)

**Reply by planger** [reply]
I've updated the link, sorry, github seemed to have cut it off.

**Reply by wss29** [reply]
Hi,
   I have tried the above link svg foreign object demo, and it works, but the problem is that it needs to translate to snabbdom's vnode, for simple HTML it is ok, but for complex UI components, it 's hard to translate to vnode manually.
I look at demo about foreignobject from website and find it can directly insert HTML elements inside foreign object 
![image](https://user-images.githubusercontent.com/4537809/219908409-25ef4f97-0b87-4de4-a5fb-fcd8a083d0c7.png)
but why can not insert HTML element into foregnobject in glsp node view,
